### ISSUE TITLE
Coherence key error on held out set 

### ISSUE THREAD

**Issue by ihavemanyquestions** [RaRe-Technologies/gensim#3111]
Hi, I am trying to train a LDA model on a train set (80% of the data):

```
lda_model = gensim.models.ldamulticore.LdaMulticore(corpus=corpus_train,
                                                                id2word=id2word_train,
                                                                 num_topics=n_topics, 
                                                                 random_state=100,
                                                                 chunksize=200,
                                                                 workers=20,
                                                                 iterations=100)
```

And get the coherence on a test set (20% held out test):

```
cv_coherence_model_lda = CoherenceModel(model=lda_model, texts=texts_test, dictionary=id2word_test, coherence='c_v')
cv_coherence_total = cv_coherence_model_lda.get_coherence()
```

I run this for n_topics from 1 to 30. 
However, it keeps crashing on random n_topics, with the error outlined below. The Key Error number changes all the time. Any help would be appreciated.

```
<ipython-input-8-8b4757057a4f> in find_optimal_topics_gensim(data, splits, test_size)
     87             #coherence models
     88             cv_coherence_model_lda = CoherenceModel(model=lda_model, texts=texts_test, dictionary=id2word_test, coherence='c_v')
---> 89             cv_coherence_total = cv_coherence_model_lda.get_coherence()
     90 

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/gensim/models/coherencemodel.py in get_coherence(self)
    610 
    611         """
--> 612         confirmed_measures = self.get_coherence_per_topic()
    613         return self.aggregate_measures(confirmed_measures)
    614 

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/gensim/models/coherencemodel.py in get_coherence_per_topic(self, segmented_topics, with_std, with_support)
    570             segmented_topics = measure.seg(self.topics)
    571         if self._accumulator is None:
--> 572             self.estimate_probabilities(segmented_topics)
    573 
    574         kwargs = dict(with_std=with_std, with_support=with_support)

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/gensim/models/coherencemodel.py in estimate_probabilities(self, segmented_topics)
    542                 kwargs['model'] = self.keyed_vectors
    543 
--> 544             self._accumulator = self.measure.prob(**kwargs)
    545 
    546         return self._accumulator

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/gensim/topic_coherence/probability_estimation.py in p_boolean_sliding_window(texts, segmented_topics, dictionary, window_size, processes)
    152         accumulator = WordOccurrenceAccumulator(top_ids, dictionary)
    153     else:
--> 154         accumulator = ParallelWordOccurrenceAccumulator(processes, top_ids, dictionary)
    155     logger.info("using %s to estimate probabilities from sliding windows", accumulator)
    156     return accumulator.accumulate(texts, window_size)

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/gensim/topic_coherence/text_analysis.py in __init__(self, processes, *args, **kwargs)
    429 
    430     def __init__(self, processes, *args, **kwargs):
--> 431         super(ParallelWordOccurrenceAccumulator, self).__init__(*args)
    432         if processes < 2:
    433             raise ValueError(

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/gensim/topic_coherence/text_analysis.py in __init__(self, relevant_ids, dictionary)
    285 
    286         """
--> 287         super(WindowedTextsAnalyzer, self).__init__(relevant_ids, dictionary)
    288         self._none_token = self._vocab_size  # see _iter_texts for use of none token
    289 

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/gensim/topic_coherence/text_analysis.py in __init__(self, relevant_ids, dictionary)
    188         """
    189         super(UsesDictionary, self).__init__(relevant_ids)
--> 190         self.relevant_words = _ids_to_words(self.relevant_ids, dictionary)
    191         self.dictionary = dictionary
    192         self.token2id = dictionary.token2id

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/gensim/topic_coherence/text_analysis.py in _ids_to_words(ids, dictionary)
     59     top_words = set()
     60     for word_id in ids:
---> 61         word = dictionary.id2token[word_id]
     62         if isinstance(word, set):
     63             top_words = top_words.union(word)

KeyError: 1094
```

Versions used

#### Versions
```

Darwin-20.3.0-x86_64-i386-64bit
Python 3.7.9 (v3.7.9:13c94747c7, Aug 15 2020, 01:31:08) 
[Clang 6.0 (clang-600.0.57)]
Bits 64
NumPy 1.19.1
SciPy 1.5.2
gensim 4.0.1
FAST_VERSION 0
```

**Comment by piskvorky** [comment]
Thanks for reporting.

I think this is a duplicate of #2711, hopefully resolved in #2830.

**Comment by ihavemanyquestions** [comment]
Hi, thank you for quickly picking this up and being so quick to have another look at this! 

I tried to test the latest update (by @pietrotrope from 2 days ago) from #2830  today on a few of my datasets, and it still randomly throws the same error I outlined above , so given the fix I am not sure why it would keep happening; or whether my bug is of a different nature.

**Comment by ihavemanyquestions** [comment]
Hi the fix from #2830 did work; so i can say my current issue can be closed! thank you for this. 

My (new) problem now seems to be linked to #3040 , so I will move on to seeing potential fixes for that.
